src/main/java/com/google/devtools/common/options/Converters.java

```diff
@@ -625,4 +635,73 @@ public String getTypeDescription() {
       return "Converts to a CacheBuilderSpec, or null if the input is empty";
     }
   }
+
+  public static class ExpressionConverter implements Converter<Expression> {
```

Comment:
Do you mean that ExpressionConverter should be moved to [LoadingOptions](https://github.com/bazelbuild/bazel/blob/1fe12e26a4b3d898326385a05b15b9a4408d5454/src/main/java/com/google/devtools/build/lib/pkgcache/LoadingOptions.java)? 
The previous solution used CommaSeparatedOptionListConverter which is also a part of [Converters.java](https://github.com/bazelbuild/bazel/blob/1fe12e26a4b3d898326385a05b15b9a4408d5454/src/main/java/com/google/devtools/common/options/Converters.java#L292) so I do not see why this new converter would not belong there aswell.

Reply:
> Do you mean that ExpressionConverter should be moved to LoadingOptions?

Yes, exactly.

> The previous solution used CommaSeparatedOptionListConverter which is also a part of Converters.java so I do not see why this new converter would not belong there as well.

The reason is that Converters is for commonly used converters, but this one is very esoteric and has only a single user (and is likely to have only a single user), so the appropriate place for the complexity is adds is with that user, not with Converters.

Reply:
Making this class private will not work since the methods **convert** and **getTypeDescription** must be accessible outside of **LoadingOptions**. Do you still want it to be moved but stay public or is it better if it stays in **Converters**?